### ISSUE TITLE
Image Items in MasonryGridView moving around

### ISSUE THREAD

**Issue by codeOfJannik** [letsar/flutter_staggered_grid_view#222]
I've implemented a MasonryGridView for images like this:
```dart
Widget staggeredGridView() => MasonryGridView.count(
        crossAxisCount: 2,
        itemCount: widget.items.length,
        mainAxisSpacing: 4.0,
        crossAxisSpacing: 4.0,
        itemBuilder: (BuildContext context, int index) => InkResponse(
          splashColor: Colors.blue,
          onTap: () => _openItem(widget.items[index]),
          child: ItemCell(
            item: widget.items[index],
            image: BlocProvider.of<StorageCubit>(context)
                .getImage(widget.items[index], ImageType.small),
          ),
        ),
      );
```

The ItemCell implementation looks like this:
```dart

@override
  Widget build(BuildContext context) => Container(
        color: Colors.white,
        child: FutureBuilder<Image?>(
            future: widget.image,
            builder: (BuildContext context, AsyncSnapshot<Image?> snapshot) {
              final image = snapshot.data?.image;
              if (snapshot.hasData && image != null) {
                return Image(
                  image: image,
                  fit: BoxFit.cover,
                );
              }

              // Empty state
              if (widget.item.imageFileIds.isEmpty) {
                return Column(
                  children: [
                    LayoutBuilder(
                      builder: (context, constraints) => Container(
                          padding: const EdgeInsets.all(16),
                          child: Image.asset(
                            "assets/images/item_placeholder.png",
                            width: constraints.maxWidth / 3,
                          )),
                    ),
                    Row(
                      children: [
                        Container(
                          padding: const EdgeInsets.only(left: 4),
                          child: Text(
                            widget.item.title ?? "",
                            textAlign: TextAlign.left,
                          ),
                        ),
                      ],
                    ),
                  ],
                );
              }

              // Loading indicator
              return SizedBox(
                width: 200,
                height: 200,
                child: Container(
                  alignment: Alignment.center,
                  child: const CircularProgressIndicator(
                    color: Colors.blue,
                  ),
                ),
              );
            }),
      );
```

The implementation works fine for scrolling downwards but when I scroll upwards the grid items are moving around and flickering. Could you please check if that's an issue on my side. Is there anything I could change to avoid these flickering? I already tried passing different properties to count() but nothing helped.

Thanks for a short response.

Video of the current implementation state.
https://user-images.githubusercontent.com/35294257/148058465-147e1ec7-96d4-49b8-8129-abafdda58820.mp4

**Comment by letsar** [comment]
Can you provide an example I would just have to run?

**Comment by codeOfJannik** [comment]
I resolved the issue by myself. The problem was the use of the Future<Image> with the SizedBox. Each time when the user scrolled upwards the cells have been re-rendered and since there was a short moment where snapshot.hasData && image != null was false, even when the image file was already available on the local storage. So each time a cell was re-rendered the loading indicator in the 200x200 SizedBox was shown before the actual image was rendered.

I changed my implementation to have something like an image cache where I store my image widgets one they have been rendered, so I can use them in the cell when the cell is re-rendered.
